### PR TITLE
Add overloads (multiple signature displayment) support

### PR DESCRIPTION
**Pull request by hit9**
Before this change:

image

After this change:

image

### CODE REVIEW

File: lua/sig-window-nvim.lua [hunk 1]
```diff
@@ -69,18 +78,22 @@ local function close_signature_window(bufnr)
...
 end
 
-local function show_signature_window(label, active_ix_start, active_ix_end, config)
+local function show_signature_window(label, active_ix_start, active_ix_end, config, other_labels)
   local bufnr = vim.api.nvim_get_current_buf()
   local w_bufnr = vim.api.nvim_create_buf(false, true)
-  vim.api.nvim_buf_set_lines(w_bufnr, 0, -1, true, {label})
+
+  table.insert(other_labels, 1, label)  -- put activeSignature label the first line
+  vim.api.nvim_buf_set_lines(w_bufnr, 0, -2, true, other_labels)
```
Comment (erhickey) [inline lua/sig-window-nvim.lua]: Why change the 3rd argument (end) to nvim_buf_set_lines from -1 to -2?

File: lua/sig-window-nvim.lua [hunk 2]
```diff
@@ -69,18 +78,22 @@ local function close_signature_window(bufnr)
...
   highlight_text(w_bufnr, active_ix_start, active_ix_end, config.hl_group)
 
-  local width, height = calc_window_dimensions(label, config.max_width, config.max_height)
+  local lines = other_labels
```
Comment (erhickey) [inline lua/sig-window-nvim.lua:89]: Is this lines variable necessary?

File: lua/sig-window-nvim.lua [hunk 3]
```diff
@@ -69,18 +78,22 @@ local function close_signature_window(bufnr)
...
+  local width, height = calc_window_dimensions(lines, config.max_width, config.max_height)
   local winnr = vim.api.nvim_open_win(w_bufnr, false, window_config(label, config, width, height))
```
Comment (erhickey) [inline lua/sig-window-nvim.lua:91]: I think window_config should be updated to take other_labels (without the active label) as the last argument, but that doesn't need to be done in this PR. I will take that change on and update the documentation.

File: lua/sig-window-nvim.lua [hunk 4]
```diff
@@ -29,7 +36,9 @@ local function highlight_text(bufnr, start_ix, end_ix, highlight_group)
   end
 end
 
-local function calc_window_dimensions(text, max_width, max_height)
+local function calc_window_dimensions(lines, max_width, max_height)
+  text = table.concat(lines, "\n")
```
Comment (erhickey) [inline lua/sig-window-nvim.lua:40]: I think this is over-simplifying the problem. Each line should be checked to determine the appropriate width and height. I'm ok with fixing this after merging the PR, I think this simple solution is sufficient for now. Feel free to leave this as is or take a stab at it.

**Review by erhickey — CHANGES_REQUESTED** [review]
Please update the branch/PR so the commit 16bc5e4 ([Set filetype to floating window's buffer from parent buffer's filetype](https://github.com/erhickey/sig-window-nvim/pull/2/commits/16bc5e4e911bd7f9cf491d360985a097403a88e9)) is not part of it.

File: lua/sig-window-nvim.lua (hunk 1 above)
Comment (hit9) [reply lua/sig-window-nvim.lua]: oh.. this is my typo changes.

**Comment by hit9** [comment]
> Please update the branch/PR so the commit [16bc5e4](https://github.com/erhickey/sig-window-nvim/commit/16bc5e4e911bd7f9cf491d360985a097403a88e9) ([Set filetype to floating window's buffer from parent buffer's filetype](https://github.com/erhickey/sig-window-nvim/pull/2/commits/16bc5e4e911bd7f9cf491d360985a097403a88e9)) is not part of it.

Forced pushed now. this commit was removed.

**Review by erhickey — APPROVED** [review]
looks good, thanks for the updates and the contribution

**Pull request merged**